src/NHibernate/Cache/StandardQueryCache.cs

```diff
@@ -492,6 +572,10 @@ public void Destroy()
 			IList cacheable)
 		{
 			Log.Debug("returning cached query results for: {0}", key);
+
+			if (!HasResults(cacheable))
+				return new List<object>();
```

Comment:
That short-circuit could be removed, as the methods called here also handle it. (These other methods needs to handle it themselves for the GetMany case.)